### PR TITLE
[#3] Set up network layer to make API requests and parse response

### LINKED ISSUE #3: Set up network layer to make API requests and parse response
# Why
To make the network requests to the API endpoints, we have to set up the network layer accordingly.

# Acceptance criteria
- [x] Retrieve the host URL, app ID, and client secret from `.env` and `.env.staging`.
- [x] Set up DI

### PR DESCRIPTION
**Pull request by AVI5HEK**
- Close #3 

## What happened 👀

* Retrieve the host URL, app ID, and client secret from .env and .env.staging.
* Set up DI.

## Insight 📝

* Set up the credentials retrieval from the env files and DI.
* Injected `CredentialRepository` into the app to call the endpoint and get the result as shown in the PoW.

**Note: The host URL, endpoints of `ApiService`, and the implementation of `CredentialRepository` will be replaced with actual endpoints for our project later. This is used only to create the PoW for the network setup.**

## Proof Of Work 📹

image

### CODE REVIEW

**Review by Wadeewee — APPROVED** [review]
LGTM 🚀

File: lib/di/provider/dio_provider.dart [hunk 1]
```diff
@@ -31,8 +33,8 @@ class DioProvider {
     }
 
     return dio
-      ..options.connectTimeout = 3000 as Duration?
-      ..options.receiveTimeout = 5000 as Duration?
+      ..options.connectTimeout = const Duration(milliseconds: 30000)
```
Comment (markgravity) [inline lib/di/provider/dio_provider.dart:36]: How about using `const Duration(seconds: 30)` here 🙏

File: lib/di/provider/dio_provider.dart [hunk 2]
```diff
@@ -31,8 +33,8 @@ class DioProvider {
...
+      ..options.receiveTimeout = const Duration(milliseconds: 50000)
```
Comment (markgravity) [inline lib/di/provider/dio_provider.dart:37]: Same here

File: lib/main.dart [hunk 3]
```diff
@@ -80,7 +83,7 @@ class HomeScreen extends StatelessWidget {
             const SizedBox(height: 24),
             Text(AppLocalizations.of(context)!.hello),
             Text(
-              FlutterConfig.get('SECRET'),
+              FlutterConfig.get('CLIENT_SECRET'),
```
Comment (markgravity) [inline lib/main.dart:86]: Can we use `env. authClientSecret` here, please 🙏

File: lib/di/provider/dio_provider.dart (hunk 1 above)
Comment (AVI5HEK) [reply lib/di/provider/dio_provider.dart]: Resolved in https://github.com/nimblehq/ic-flutter-avishek/pull/28/commits/55417b71c60b16af19eb96ca7bb0016f46388dc0.

File: lib/di/provider/dio_provider.dart (hunk 2 above)
Comment (AVI5HEK) [reply lib/di/provider/dio_provider.dart]: Resolved in https://github.com/nimblehq/ic-flutter-avishek/pull/28/commits/55417b71c60b16af19eb96ca7bb0016f46388dc0.

File: lib/main.dart (hunk 3 above)
Comment (AVI5HEK) [reply lib/main.dart]: Resolved in https://github.com/nimblehq/ic-flutter-avishek/pull/28/commits/55417b71c60b16af19eb96ca7bb0016f46388dc0.

**Review by markgravity — APPROVED** [review]

**Pull request merged**